src/fetch.rs

```diff
@@ -67,152 +79,76 @@ fn version_is_stable(version: &CrateVersion) -> bool {
-    let versions: Versions = json::from_str(
-        r#"{
-      "versions": [
-        {
-          "crate": "foo",
-          "num": "0.6.0-alpha",
-          "yanked": false
-        },
-        {
-          "crate": "foo",
-          "num": "0.5.0",
-          "yanked": false
-        }
-      ]
-    }"#,
-    )
-    .expect("crate version is correctly parsed");
-
-    assert_eq!(
-        read_latest_version(&versions, false)
-            .unwrap()
-            .version()
-            .unwrap(),
-        "0.5.0"
-    );
-}
-
-#[test]
-fn get_latest_unstable_or_stable_version_from_json() {
-    let versions: Versions = json::from_str(
-        r#"{
-      "versions": [
-        {
-          "crate": "foo",
-          "num": "0.6.0-alpha",
-          "yanked": false
-        },
-        {
-          "crate": "foo",
-          "num": "0.5.0",
-          "yanked": false
-        }
-      ]
-    }"#,
-    )
-    .expect("crate version is correctly parsed");
-
-    assert_eq!(
-        read_latest_version(&versions, true)
-            .unwrap()
-            .version()
-            .unwrap(),
-        "0.6.0-alpha"
-    );
-}
-
-#[test]
-fn get_latest_version_from_json_test() {
-    let versions: Versions = json::from_str(
-        r#"{
-      "versions": [
-        {
-          "crate": "treexml",
-          "num": "0.3.1",
-          "yanked": true
-        },
-        {
-          "crate": "treexml",
-          "num": "0.3.0",
-          "yanked": false
-        }
-      ]
-    }"#,
-    )
-    .expect("crate version is correctly parsed");
-
-    assert_eq!(
-        read_latest_version(&versions, false)
-            .unwrap()
-            .version()
-            .unwrap(),
-        "0.3.0"
-    );
-}
-
-#[test]
-fn get_no_latest_version_from_json_when_all_are_yanked() {
-    let versions: Versions = json::from_str(
-        r#"{
-      "versions": [
-        {
-          "crate": "treexml",
-          "num": "0.3.1",
-          "yanked": true
-        },
-        {
-          "crate": "treexml",
-          "num": "0.3.0",
-          "yanked": true
-        }
-      ]
-    }"#,
-    )
-    .expect("crate version is correctly parsed");
-
-    assert!(read_latest_version(&versions, false).is_err());
+fn update_git_repo(path: impl AsRef<Path>, url: &Url) -> Result<()> {
+    let path = path.as_ref();
+    let repo = git2::Repository::open(path)?;
+    let colorchoice = if atty::is(atty::Stream::Stdout) {
+        ColorChoice::Auto
+    } else {
+        ColorChoice::Never
+    };
+    let mut output = StandardStream::stdout(colorchoice);
+    output.set_color(ColorSpec::new().set_fg(Some(Color::Green)).set_bold(true))?;
+    write!(output, "{:>12}", "Updating")?;
+    output.reset()?;
+    writeln!(output, " '{}' index", url)?;
+
+    repo.remote_anonymous(url.as_str())?.fetch(
+        &["refs/heads/master:refs/remotes/origin/master"],
+        None,
+        None,
+    )?;
+
+    Command::new("git").arg("gc").current_dir(path).output()?;
```

Comment:
hmm, I'm not if it's a good idea to call `git gc` on user's local registry
@killercup what are you thoughts on this?

Reply:
You are right, I'm going to remove this line.

Reply:
done.